packages/peregrine/lib/talons/AuthModal/useAuthModal.js

```diff
@@ -63,12 +63,19 @@ export const useAuthModal = props => {
     const handleSignOut = useCallback(async () => {
         // After logout, reset the store to set the bearer token.
         // https://www.apollographql.com/docs/react/networking/authentication/#reset-store-on-logout
-        await resetStore();
         await signOut({ revokeToken });
 
-        // Refresh this page.
+        // Pause automatic persistence.
+        client.persistor.pause();
+        // Delete everything in the storage provider.
+        await client.persistor.purge();
+
+        // Wipe the apollo store. Do not replay queries.
+        await client.clearStore();
```

Comment:
Instead of clearStore/refresh we should be able to just call `resetStore` which replays any queries. Unfortunately this replays await queries such as `getCartDetails` that use the old cart id and results in a thrown auth error. I'd love to figure this out, but it really only matters if we don't want to refresh the page. You can see the error for yourself if you replace lines 74-77 with the below code.

```js
await client.resetStore();
client.persistor.resume();
closeDrawer();
 ```

Reply:
We may be able to replace this with specific cache deletion.